compiler/rustc_hir_analysis/src/collect/generics_of.rs

```diff
@@ -249,6 +249,10 @@ pub(super) fn generics_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::Generics {
     // Now create the real type and const parameters.
     let type_start = own_start - has_self as u32 + params.len() as u32;
     let mut i = 0;
+    let mut next_index = || {
+        let new_i = i + 1;
+        std::mem::replace(&mut i, new_i) as u32 + type_start
```

Comment:
:sweat_smile: i guess that works '^^ feels a lot less readable than `let prev = i; i += 1; i + type_start` to me.

not blocking, so your choice whether we should merge this, r=me